### ISSUE TITLE
Regex validation for named parameters in routes

### ISSUE THREAD

**Issue by antitoxic** [fuel/core#702]
It would be nice if the developer has the ability to set a regular expression for the named parameters in the url
For instance:

``` php
<?php
return array(
    'users/view/:id(:num)' => 'users/view',
    // or alternatively...
    'users/view/:id([0-9])' => 'users/view',
    // or alternatively...
    'users/view/:id' => array( 'users/view', array(':id'=>':num'))
);
```

If the named parameter doesn't match the regular expression in the brackets then continue looking for a matching route. Even if the syntax won't be the same - just to have the ability to restrict this.

I know currently `:<parameterName>` matches an entire url segment but this feature will save some **repetative** code writing:
- no check for the parameter or at least less checks
- no need to fire 404 manually
- better separation of logic as we ultimately don't want not to enter the controller if the route doesn't really match

**Comment by tmihalik** [comment]
try this:

``` php
return array(
    'users/view/(?=:num):id' => 'users/view',
);
```

Can somebody tell me how to define default value for :id ?

**Comment by antitoxic** [comment]
@tmihalik ?
This doesn't actually restrict the named parameter. My example above is just a use case.
The `id` may be more than 1 character and then your solution is not working. I'm working on a modification which will enable such restriction. I just don't know what is the best syntax for the definition in routes.php?

I'm thinking about:

```
'users/view/:id' => array( 'users/view', array(':id'=>':num'))
```

will be most suitable
Any suggestions?

**Comment by tmihalik** [comment]
Its gonna be ok.

**Comment by aar0nr** [comment]
+1

**Comment by antitoxic** [comment]
I've implemented such feature with #708
